### PR TITLE
DOC: Clarify display.precision

### LINKED ISSUE #21004: (Further) clarify documentation for display.precision
The generated "stable" [docs for `pandas.set_option()`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.set_option.html) (as well as the [Travis docs](https://pandas-docs.github.io/pandas-docs-travis/generated/pandas.set_option.html) as of this writing) describe `display.precision` as:

> Floating point output precision (number of significant digits)

However, the “[Options and Settings](http://pandas.pydata.org/pandas-docs/version/0.22/options.html)“  docs for describe `display.precision` as

> Floating point output precision in terms of number of places after the decimal

The second description appears to be more accurate (as well as newer; compare for instance the still-Google-popular [0.15 docs](http://pandas.pydata.org/pandas-docs/version/0.15/options.html)).

#### Code Sample

```python
import pandas as pd
df = pd.DataFrame([[1234.0, 123.4, 12.34, 1.234, .1234, .01234, .001234]])
df
#         0      1      2      3       4        5         6
# 0  1234.0  123.4  12.34  1.234  0.1234  0.01234  0.001234
pd.options.display.precision = 1
df
#         0      1     2    3    4        5        6
# 0  1234.0  123.4  12.3  1.2  0.1  1.2e-02  1.2e-03
```

#### Expected Output

The output above clearly isn't significant digits, which (using numpy's [`format_float_positional()`](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.format_float_positional.html)) would look something more like:

```python
import numpy as np
df.applymap(lambda x: np.format_float_positional(x, 1, fractional=False).rstrip('.'))
#       0    1   2  3    4     5      6
# 0  1000  100  10  1  0.1  0.01  0.001
```

#### Problem description

The generated documentation for `set_option()` is inconsistent with the (hand-written?) “Options and Settings” doc. It’s not obvious which is correct until either you try it yourself, or dig through enough old docs to realize the “Options and Settings” doc has been more recently updated.

That said, "Number of places after the decimal" is also somewhat unclear, as it's not obvious why `1234.0` is `1234.0` rather than `1.2e+03`, or why `123.4` isn't `1.2e+02`. From experimentation, it looks like pandas decides this based on the entire series, rather than the individual value, so that if I transpose the frame, I do get a consistent one digit after the decimal point:

```python
df.transpose()
#          0
# 0  1.2e+03
# 1  1.2e+02
# 2  1.2e+01
# 3  1.2e+00
# 4  1.2e-01
# 5  1.2e-02
# 6  1.2e-03
```

I'm sure this is documented somewhere, but it's not all that intuitive, and it would be nice if that document was linked from both the `set_option()` and “Options and Settings” docs for `display.precision`.

(Also, it would be nice if pandas had a display option for significant digits. :) ) 

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Darwin
OS-release: 17.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: None
pip: 10.0.1
setuptools: 39.1.0
Cython: None
numpy: 1.14.3
scipy: 1.0.1
pyarrow: None
xarray: None
IPython: 6.3.1
sphinx: None
patsy: None
dateutil: 2.7.2
pytz: 2018.4
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by hongshaoyang**
- [x] closes #21004
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by rhshadrach — CHANGES_REQUESTED** [review]
This change seems less precise than what is in Settings and Options. Can we just copy that? 

> Floating point output precision in terms of number of places after the decimal, for regular formatting as well as scientific notation. Similar to numpy’s precision print option

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -85,8 +85,8 @@ def use_numba_cb(key):
 
 pc_precision_doc = """
 : int
-    Floating point output precision (number of significant digits). This is
-    only a suggestion
+    Floating point output precision (in terms of decimal places). Similar to
+    numpy's ``precision`` print option.
```
Comment (jreback) [inline pandas/core/config_init.py:89]: you can directly reference the numpy function itself ```:meth:`np. ....` ```

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by hongshaoyang** [comment]
Thanks @jreback, @rhshadrach, updated.

**Review by rhshadrach — APPROVED** [review]

**Comment by hongshaoyang** [comment]
all checks passing.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @hongshaoyang

**Pull request merged**